src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,793 @@
+#include <cos_kernel_api.h>
+#include <sl.h>
+#include <sl_lock.h>
+
+#include "cFE_util.h"
+#include "ostask.h"
+
+#include "gen/osapi.h"
+#include "gen/common_types.h"
+#include "gen/cfe_time.h"
+
+#define HZ_PAUSE (1000 * 1000)
+
+void
+timer_fn_1hz(void *d)
+{
+	cycles_t now, start;
+
+	rdtscll(start);
+	start += sl_usec2cyc(HZ_PAUSE);
+
+	while (1) {
+		rdtscll(now);
+		if (now > start) {
+			CFE_TIME_Local1HzISR(); /* input param is signum. but CFE_PSP_TimerHandler doesn't seem to use
+			                           it. */
+		}
+
+		sl_thd_block_periodic(0);
+	}
+}
+
+/*
+** Internal Task helper functions
+*/
+
+#define MAIN_DELEGATE_THREAD_PRIORITY 2
+
+/*  We need to keep track of this to check if register or delete handler calls are invalid */
+thdid_t main_delegate_thread_id;
+
+void
+OS_SchedulerStart(cos_thd_fn_t main_delegate)
+{
+	sl_init(SL_MIN_PERIOD_US);
+
+	struct sl_thd *         main_delegate_thread = sl_thd_alloc(main_delegate, NULL);
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = MAIN_DELEGATE_THREAD_PRIORITY}};
+	sl_thd_param_set(main_delegate_thread, sp.v);
+	main_delegate_thread_id = sl_thd_thdid(main_delegate_thread);
```

Comment:
"delegate thread" is a little bit of a strange name.  Not really sure what "delegate" has to do with all of this.

Reply:
It's called `main_delegate_thread`. It's the thread we delegate main execution to. Do you have a better name in mind?